homeassistant/components/totalconnect/binary_sensor.py

```diff
@@ -0,0 +1,103 @@
+"""Interfaces with TotalConnect sensors."""
+import logging
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_SMOKE,
+    BinarySensorDevice,
+)
+
+from . import DOMAIN as TOTALCONNECT_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up a sensor for a TotalConnect device."""
+    if discovery_info is None:
+        return
+
+    sensors = []
+
+    client_locations = hass.data[TOTALCONNECT_DOMAIN].client.locations
+
+    for k in client_locations:
+        for zone in client_locations[k].zones:
+            sensors.append(TotalConnectBinarySensor(zone, k, client_locations))
+    add_entities(sensors)
+
+
+class TotalConnectBinarySensor(BinarySensorDevice):
+    """Represent an TotalConnect zone."""
+
+    def __init__(self, zone_id, location_id, locations):
+        """Initialize the TotalConnect status."""
+        self._zone_id = zone_id
+        self._location_id = location_id
+        self._name = "TC {} zone {}".format(
+            locations[location_id].location_name, zone_id
+        )
+        self._unique_id = "TC {} zone {}".format(
+            locations[location_id].location_name, zone_id
+        )
```

Comment:
Why are name and unique_id the same?

Reply:
I changed it so they are now different from each other.  I also fixed unique_id so it is more likely to remain unchanged (the location_name is user changeable from within total-connect-client / Honeywell system).